packages/itwin/grouping-mapping-widget/src/widget/components/GroupingMapping.tsx

```diff
@@ -69,13 +41,21 @@ const GroupingMapping = ({ accessToken, prefix, client }: GroupingMappingProps)
   }, [clientProp]);
 
   return (
-    currentAccessToken ? <ApiContext.Provider value={{ accessToken: currentAccessToken, prefix }}>
+    <GroupingMappingApiConfigContext.Provider
+      value={{
+        getAccessToken:
+          getAccessToken ??
+          (async () =>
```

Comment:
I think this might cause excessive re-renders when no `getAccessToken` is provided because each time a new function (lambda expression) will be created. Do you know if multiple instances of this lambda will have reference equality? If not, maybe a non-lambda function would have reference equality (non exported function could be defined in the same file).

Reply:
You are correct. The lambda will not have reference equality per render. I wrapped it in a useCallback. I also realized that the array dependencies could be more specific.

Reply:
I'll put the object in the value in a useState. Earlier, nothing was using the object as a dependency within the widget. Only the values within. Even if the object was recreated the properties would be unchanged and no rerender would happen. But I don't mind being safe about it.

Reply:
I put the object in the value in a useState. Nevermind my earlier comment.